kaniko.go

```diff
@@ -174,6 +176,50 @@ func (p Plugin) Exec() error {
 		return fmt.Errorf("repository name to publish image must be specified")
 	}
 
+	if p.Build.LocalTarPath != "" {
+		if p.Build.Repo == "" {
+			return fmt.Errorf("destination repository name must be specified")
+		}
+
+		// Validate local tarball exists
+		if _, err := os.Stat(p.Build.LocalTarPath); os.IsNotExist(err) {
+			return fmt.Errorf("local tarball does not exist at path: %s", p.Build.LocalTarPath)
+		}
+
+		var tags []string
+		var err error
+		if p.Build.AutoTag {
+			tags, err = p.Build.AutoTags()
+			if err != nil {
+				return err
+			}
+		} else if len(p.Build.Tags) > 0 {
+			tags = p.Build.Tags
+		} else {
+			tags = []string{"latest"}
+		}
+
+		// Load the image from the tarball
+		img, err := crane.Load(p.Build.LocalTarPath)
+		if err != nil {
+			return fmt.Errorf("failed to load image from tarball: %v", err)
+		}
+
+		for _, tag := range tags {
+			dest := fmt.Sprintf("%s:%s", p.Build.Repo, tag)
+
+			// Push the image to the destination
+			err := crane.Push(img, dest)
+			if err != nil {
+				return fmt.Errorf("failed to push image to %s: %v", dest, err)
+			}
+
+			fmt.Printf("Successfully pushed %s\n", dest)
```

Comment:
Let's include image name and destination in the log message.

Reply:
Included. ✅